test/test_torch.py

```diff
@@ -6088,6 +6088,101 @@ def test_pow(self, device):
             torch.pow(m1, 1, out=out)
             self.assertEqual(out, m1)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @onlyOnCPUAndCUDA
+    @dtypes(*list(product(torch.testing.get_all_dtypes(include_bool=False, include_complex=False),
+                          torch.testing.get_all_dtypes(include_bool=False, include_complex=False))))
+    def test_float_power(self, device, dtypes):
+        def to_np(value):
+            if isinstance(value, torch.Tensor):
+                if value.dtype == torch.bfloat16:
+                    value = value.to(dtype=torch.float)
+                return value.cpu().numpy()
+            return value
+
+        base_dtype = dtypes[0]
+        exp_dtype = dtypes[1]
+
+        base = make_tensor((30,), device, base_dtype, low=-100, high=100)
+        base[0] = base[3] = base[7] = 0
+        exp = make_tensor((30,), device, exp_dtype, low=-2, high=2)
+        exp[0] = exp[4] = exp[6] = 0
+
+        expected = torch.from_numpy(np.float_power(to_np(base), to_np(exp)))
+
+        for op in (torch.float_power, torch.Tensor.float_power, torch.Tensor.float_power_):
+            result = op(base.clone(), exp)
+            self.assertEqual(expected, result)
+
+            if op is torch.float_power:
+                out = torch.empty_like(base).to(device=device, dtype=torch.float64)
+                op(base, exp, out=out)
+                self.assertEqual(expected, out)
+
+            for i in range(10):
+                expected_scalar_exp = np.float_power(to_np(base), i)
+                expected_scalar_base = np.float_power(i, to_np(exp))
+
+                result = op(base.clone(), i)
+                self.assertEqual(expected_scalar_exp, result)
+
+                if op is torch.float_power:
+                    result = op(i, exp)
+                    self.assertEqual(expected_scalar_base, result)
+
+                    out = torch.empty_like(base).to(device=device, dtype=torch.float64)
+                    op(base, i, out=out)
+                    self.assertEqual(expected_scalar_exp, out)
+
+                    out = torch.empty_like(exp).to(device=device, dtype=torch.float64)
+                    op(i, exp, out=out)
+                    self.assertEqual(expected_scalar_base, out)
+
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @onlyOnCPUAndCUDA
+    @dtypes(*(torch.testing.get_all_dtypes(include_bool=False)))
+    def test_float_power_with_complex(self, device, dtype):
```

Comment:
I think this variant of the test can be merged with the above.

Reply:
I merged this variant of test with the above. But as mentioned above, there is another problem.